auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -244,6 +253,114 @@ public void onClick(View view) {
+                    email.init(null);
+                    provider = email;
+
+                    break;
+                case PhoneAuthProvider.PROVIDER_ID:
+                    PhoneSignInHandler phone = supplier.get(PhoneSignInHandler.class);
+                    phone.init(idpConfig);
+                    provider = phone;
+
+                    break;
+                case AuthUI.ANONYMOUS_PROVIDER:
+                    AnonymousSignInHandler anonymous = supplier.get(AnonymousSignInHandler.class);
+                    anonymous.init(getFlowParams());
+                    provider = anonymous;
+
+                    break;
+                default:
+                    throw new IllegalStateException("Unknown provider: " + providerId);
+            }
+            mProviders.add(provider);
+
+            provider.getOperation().observe(this, new ResourceObserver<IdpResponse>(this) {
+                @Override
+                protected void onSuccess(@NonNull IdpResponse response) {
+                    handleResponse(response);
+                }
+
+                @Override
+                protected void onFailure(@NonNull Exception e) {
+                    handleResponse(IdpResponse.from(e));
+                }
+
+                private void handleResponse(@NonNull IdpResponse response) {
+                    if (!response.isSuccessful()) {
+                        // We have no idea what provider this error stemmed from so just forward
+                        // this along to the handler.
+                        handler.startSignIn(response);
+                    } else if (AuthUI.SOCIAL_PROVIDERS.contains(providerId)) {
+                        // Don't use the response's provider since it can be different than the one
+                        // that launched the sign-in attempt. Ex: the email flow is started, but
+                        // ends up turning into a Google sign-in because that account already
+                        // existed. In the previous example, an extra sign-in would incorrectly
+                        // started.
+                        handler.startSignIn(response);
+                    } else {
+                        // Email or phone: the credentials should have already been saved so
+                        // simply move along. Anononymous sign in also does not require any
+                        // other operations.
+                        finish(response.isSuccessful() ? RESULT_OK : RESULT_CANCELED,
+                                response.toIntent());
+                    }
+                }
+            });
+
+            View loginButton = findViewById(buttonId);
+            loginButton.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    provider.startSignIn(AuthMethodPickerActivity.this);
+                }
+            });
+            mProviderHolder.addView(loginButton);
```

Comment:
I don't think we should do this, right?  In a custom layout are we responsible for making sure the right buttons are visible or are they?

Reply:
Yes. My bad, the custom layout should be responsible in making sure those are visible. We just add the OnClickListener